File: leetcode/0168-excel-sheet-column-title/0168-excel-sheet-column-title.java
class Solution {
    public String convertToTitle(int columnNumber) {
        int n = columnNumber;
        StringBuilder c = new StringBuilder();

        if (n < 27) {
            return Character.toString((char) (n + 64));
        }

        while (n > 0) {
            if (n % 26 == 0) {
                c.append('Z');
                n = n / 26;
                n--;
            } else {
                c.append((char) (n % 26 + 64));
                n = n / 26;
            }
        }

        return c.reverse().toString();
    }
}

File: leetcode/0168-excel-sheet-column-title/0168-excel-sheet-column-title.ts
function convertToTitle(columnNumber: number): string {
    let n: number = columnNumber;
    let result: string = "";

    if (n < 27) {
        return String.fromCharCode(n + 64);
    }

    while (n > 0) {
        if (n % 26 === 0) {
            result += "Z";
            n = Math.floor(n / 26);
            n--;
        } else {
            result += String.fromCharCode((n % 26) + 64);
            n = Math.floor(n / 26);
        }
    }

    return result.split("").reverse().join("");
}